CMakeLists.txt

```diff
@@ -198,124 +193,197 @@ else()
+
+add_executable(picolog_t picolog/picolog.c)
 
 target_link_libraries(picolog_t
-    picoquic-log
-    picoquic-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
-    ${CMAKE_DL_LIBS}
-    ${CMAKE_THREAD_LIBS_INIT}
-)
-
-target_include_directories(picolog_t PRIVATE loglib)
+    PRIVATE
+        #${PTLS_LIBRARIES}
+        #${OPENSSL_LIBRARIES}
+        #${CMAKE_DL_LIBS}
+        picoquic-log
+        picoquic-core)
 
-add_executable(picoquic_ct picoquic_t/picoquic_t.c
-    ${PICOQUIC_TEST_LIBRARY_FILES}
-)
+target_include_directories(picolog_t
+    PRIVATE
+        #picoquictest
+        loglib)
 
-target_link_libraries(picoquic_ct
-    picoquic-log
-    picoquic-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
-    ${CMAKE_DL_LIBS}
-    ${CMAKE_THREAD_LIBS_INIT}
-)
+set_picoquic_build_options(picolog_t)
 
-target_include_directories(picoquic_ct PRIVATE loglib)
+add_executable(picoquic_ct
+    picoquic_t/picoquic_t.c
+    ${PICOQUIC_TEST_LIBRARY_FILES})
 
-add_executable(picohttp_ct picohttp_t/picohttp_t.c
+target_link_libraries(picoquic_ct
+    PRIVATE
+        ${PTLS_LIBRARIES}
+        ${OPENSSL_LIBRARIES}
+        #${CMAKE_DL_LIBS}
+        picoquic-log
+        picoquic-core)
+
+target_include_directories(picoquic_ct
+    PRIVATE
+        ${PTLS_INCLUDE_DIRS}
+        ${OPENSSL_INCLUDE_DIR}
+        #loglib
+        picoquictest)
+
+set_picoquic_build_options(picoquic_ct)
+
+add_executable(picohttp_ct
+    picohttp_t/picohttp_t.c
     ${PICOQUIC_TEST_LIBRARY_FILES}
-    ${PICOHTTP_TEST_LIBRARY_FILES}
-)
+    ${PICOHTTP_TEST_LIBRARY_FILES})
 
 target_link_libraries(picohttp_ct
-    picoquic-log
-    picoquic-core
-    picohttp-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
-    ${CMAKE_DL_LIBS}
-    ${CMAKE_THREAD_LIBS_INIT}
-)
-
-target_include_directories(picohttp_ct PRIVATE loglib picohttp)
+    PRIVATE
+        #${PTLS_LIBRARIES}
+        #${OPENSSL_LIBRARIES}
+        #${CMAKE_DL_LIBS}
+        picoquic-log
+        picoquic-core
+        picohttp-core)
+
+target_include_directories(picohttp_ct
+    PRIVATE
+        ${PTLS_INCLUDE_DIRS}
+        picoquictest
+        #loglib
+        picohttp)
+
+set_picoquic_build_options(picohttp_ct)
 
 add_executable(picoquic_sample
     sample/sample.c
     sample/sample_client.c
-    sample/sample_server.c
-)
+    sample/sample_server.c)
 
 target_link_libraries(picoquic_sample
-    picoquic-log
-    picoquic-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
-    ${CMAKE_DL_LIBS}
-    ${CMAKE_THREAD_LIBS_INIT}
-)
+    PRIVATE
+        #${PTLS_LIBRARIES}
+        #${OPENSSL_LIBRARIES}
+        #${CMAKE_DL_LIBS}
+        picoquic-log
+        picoquic-core)
 
 target_include_directories(picoquic_sample PRIVATE loglib picoquic)
 
+set_picoquic_build_options(picohttp_ct)
+
+# PEJ - Should we introduce ctest and a test build option so that
+# downstream code will not have these tests built and so that
+# "make test" works as expected?
 set(TEST_EXES picoquic_ct picohttp_ct)
```

Comment:
Yes.

But also, consider that the test programs of the upstream code may end up using the test libraries of picoquic.